steps/measure.py

```diff
@@ -45,3 +50,44 @@ def get_bitstring_distribution(
 
     bitstring_distribution = backend.get_bitstring_distribution(circuit)
     save_bitstring_distribution(bitstring_distribution, "bitstring-distribution.json")
+
+
+def evaluate_ansatz_based_cost_function(
+    ansatz_specs: str,
+    backend_specs: str,
+    cost_function_specs: str,
+    ansatz_parameters: str,
+    qubit_operator: str,
+    noise_model="None",
+    device_connectivity="None",
+):
+    ansatz_parameters = load_circuit_template_params(ansatz_parameters)
+    # Load qubit op
+    operator = load_qubit_operator(qubit_operator)
+    ansatz_specs = json.loads(ansatz_specs)
+    if ansatz_specs["function_name"] == "QAOAFarhiAnsatz":
+        ansatz = create_object(ansatz_specs, cost_hamiltonian=operator)
```

Comment:
This is not a good solution.
Not sure if this PR is a good place to solve it, but we need to have a story for that.

Reply:
Yeah it's not great to hard code behavior for specific ansatzes here. I had just copied this from [optimize_variational_circuit](https://github.com/zapatacomputing/z-quantum-optimizers/blob/master/steps/optimizers.py#L44). Since this pattern is used elsewhere, should we just merge this PR and create a story for figuring out a solution in general?

Reply:
Yes, sounds good. Accepting it!